### PR TITLE
fix #9125

### LINKED ISSUE #9125: ties are rounded differently on c and js backends
On the c-like backends ties (values exactly between 2 adjacent integers) are rounded "away from zero".
On the js backend ties are rounded toward positive infinity.
```nim
import math
echo round(-1.5)
```
produces `-2` on the c backend, but `-1` on the js backend.

### PR DESCRIPTION
**Pull request by ringabout**
fix #9125

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -89,6 +89,8 @@
 - Added `posix_utils.osReleaseFile` to get system identification from `os-release` file on Linux and the BSDs.
   https://www.freedesktop.org/software/systemd/man/os-release.html
 
+- `math.round` now is rounded "away from zero" in JS backend which is consistent
+with other backends. see #9125. Use `-d:nimLegacyRound` for previous behavior.
```
Comment (timotheecour) [inline changelog.md]: `-d:nimLegacyJsRound`

File: tests/stdlib/tmath.nim [hunk 2]
```diff
@@ -347,6 +339,25 @@ template main =
     doAssert copySign(-NaN, 0.0).isNaN
     doAssert copySign(-NaN, -0.0).isNaN
 
+    block: # round() tests
+      # Round to 0 decimal places
+      doAssert round(54.652) ==~ 55.0
```
Comment (timotheecour) [inline tests/stdlib/tmath.nim]: wouldn't `==` work for all round tests?

Comment (ringabout) [reply tests/stdlib/tmath.nim]: >  wouldn't == work for all round tests?

I dunno, they are pre-existing.

File: lib/pure/math.nim [hunk 3]
```diff
@@ -847,7 +847,16 @@ else: # JS
   func floor*(x: float64): float64 {.importc: "Math.floor", nodecl.}
   func ceil*(x: float32): float32 {.importc: "Math.ceil", nodecl.}
   func ceil*(x: float64): float64 {.importc: "Math.ceil", nodecl.}
-  func round*(x: float): float {.importc: "Math.round", nodecl.}
+
+  when not declared(copySign) or defined(nimLegacyJsRound):
+    func round*(x: float): float {.importc: "Math.round", nodecl.}
+  else:
+    func round*[T: float64 | float32](x: T): T =
+      let magnitude = abs(x)
```
Comment (timotheecour) [inline lib/pure/math.nim]: simpler:
```nim
        result = roundImpl(x)
        if x < 0 and result - x >= 0.5: result = result - 1
```
or, (micro-optimization in the case where roundImpl would be slower than ceil in js):
```nim
        if x >= 0: result = roundImpl(x)
        else:
          result = ceil(x)
          if result - x >= 0.5: result = result - 1
```

with:
```nim
        func roundImpl(x: float): float {.importc: "Math.round", nodecl.}
```

Comment (ringabout) [reply lib/pure/math.nim]: thanks
```nim
  when (NimMajor, NimMinor) < (1, 5) or defined(nimLegacyJsRound):
    func round*(x: float): float {.importc: "Math.round", nodecl.}
  else:
    func jsRound(x: float): float {.importc: "Math.round", nodecl.}
    func round*[T: float64 | float32](x: T): T =
      if x >= 0: result = jsRound(x)
      else:
        result = ceil(x)
        if result - x >= T(0.5):
          result -= T(1.0)
```

File: tests/stdlib/tmath.nim [hunk 4]
```diff
@@ -349,6 +341,25 @@ template main =
...
+      doAssert round(Inf) == Inf
+      doAssert round(-Inf) == -Inf
+      doAssert round(NaN).isNaN
+      doAssert round(-NaN).isNaN
+      doAssert round(-0.5) ==~ -1.0
+      doAssert round(0.5) ==~ 1.0
+      doAssert round(-1.5) ==~ -2.0
+      doAssert round(1.5) ==~ 2.0
+      doAssert round(-2.5) ==~ -3.0
+      doAssert round(2.5) ==~ 3.0
```
Comment (timotheecour) [inline tests/stdlib/tmath.nim]: (superseded)
```suggestion
      doAssert round(2.5) ==~ 3.0
      doAssert round(2.5'f32) == 3.0'f32
```

Comment (timotheecour) [inline tests/stdlib/tmath.nim]:
```suggestion
    block: # round() tests
      # Round to 0 decimal places
      doAssert round(54.652) == 55.0
      doAssert round(54.352) == 54.0
      doAssert round(-54.652) == -55.0
      doAssert round(-54.352) == -54.0
      doAssert round(0.0) == 0.0
      doAssert 1 / round(0.0) == Inf
      doAssert 1 / round(-0.0) == -Inf
      doAssert round(Inf) == Inf
      doAssert round(-Inf) == -Inf
      doAssert round(NaN).isNaN
      doAssert round(-NaN).isNaN
      doAssert round(-0.5) == -1.0
      doAssert round(0.5) == 1.0
      doAssert round(-1.5) == -2.0
      doAssert round(1.5) == 2.0
      doAssert round(-2.5) == -3.0
      doAssert round(2.5) == 3.0
      doAssert round(2.5'f32) == 3.0'f32
      doAssert round(2.5'f64) == 3.0'f64
```

**Review by timotheecour — APPROVED** [review]
LGTM after last comment

**Review by timotheecour — APPROVED** [review]

File: tests/stdlib/tmath.nim [hunk 5]
```diff
@@ -176,13 +176,6 @@ block:
       doAssert round(-54.352) == -54.0
       doAssert round(0.0) == 0.0
 
-    block: # splitDecimal() tests
-      doAssert splitDecimal(54.674).intpart == 54.0
-      doAssert splitDecimal(54.674).floatpart ==~ 0.674
-      doAssert splitDecimal(-693.4356).intpart == -693.0
-      doAssert splitDecimal(-693.4356).floatpart ==~ -0.4356
-      doAssert splitDecimal(0.0).intpart == 0.0
-      doAssert splitDecimal(0.0).floatpart == 0.0
```
Comment (Clyybber) [inline tests/stdlib/tmath.nim]: Please don't remove these (I assume it's an artifact from rebasing?)

Comment (ringabout) [reply tests/stdlib/tmath.nim]: yeah

**Pull request merged**